weblog/requesthandler.go

```diff
@@ -7,26 +7,29 @@ import (
 	"strconv"
 	"strings"
 
-	"github.com/deis/logger/syslogish"
+	"github.com/deis/logger/logs"
+	"github.com/deis/logger/storage"
 	"github.com/gorilla/mux"
 )
 
 type requestHandler struct {
-	syslogishServer *syslogish.Server
+	storageAdapter storage.Adapter
 }
 
-func newRequestHandler(syslogishServer *syslogish.Server) *requestHandler {
-	return &requestHandler{syslogishServer: syslogishServer}
+func newRequestHandler(storageAdapter storage.Adapter) *requestHandler {
```

Comment:
not necessarily. for example, an `int` can implement an interface, or a struct can implement one, but all of the functions on that struct can have receivers that are not pointers (see `ImplBug` in the abocv playground link)